hansard/uk_to_dolma.py

```diff
@@ -0,0 +1,164 @@
+        elif element.tag in ["oral-heading", "minor-heading"]:
+            text = ET.tostring(element, method="text", encoding="unicode").strip()
+            parsed_text.append(f"{text}")
+        elif element.tag in [
+            "speech",
+            "ques",
+            "reply",
+            "question",
+        ]:
+            speaker = (
+                element.attrib.get("speakername", "") + ": "
+                if element.attrib.get("speakername", "")
+                else ""
+            )
+            speech_text = (
+                ET.tostring(element, method="text", encoding="unicode")
+                .strip()
+                .replace("    ", "")
+            )
+            parsed_text.append(f"{speaker + speech_text}")
+
+    return "\n\n".join(parsed_text).replace(" ", "")
+
+
+def process_files_in_folder(
+    folder_path: Path, source: str, count: bool = False
+) -> dict:
+    global COUNT
+    language = "en" if not source == "senedd-cy" else "cy"
+    date_match = re.compile(r"\d{4}-\d{2}-\d{2}")
+    for file in folder_path.iterdir():
+        if file.suffix == ".xml" and file.stem != "tmp":
+            root = ET.parse(file, parser=PARSER).getroot()
+            parsed_text = parse_hansard_xml_file(root)
+            if parsed_text:
+                if count:
+                    COUNT += len(WHITESPACE.split(parsed_text))
+                date_ = date_match.search(file.stem)
+                if date_:
+                    date = date_.group()
+                else:
+                    date = "9999-01-01"
+                yield {
+                    "id": file.stem,
+                    "text": parsed_text,
+                    "created": date,
+                    "source": f"uk-hansard-{source}",
+                    "added": str(datetime.now().date()),
+                    "metadata": {
+                        "license": str(PermissiveLicenses.OPL),
+                        "language": language,
+                        "year": date.split("-")[0],
+                    },
+                }
+
+
+def process_folder(folder_path: Path, count: bool = False) -> Iterator[dict]:
+    for subfolder in get_subfolders(folder_path):
+        if subfolder.name in FILE_NAMES:
+            source = FILE_NAMES[subfolder.name]["source"]
+            if source == "senedd":
+                for nested_subfolder in get_subfolders(subfolder):
+                    source = FILE_NAMES["senedd"][nested_subfolder.name]["source"]
+                    yield from process_files_in_folder(
+                        nested_subfolder, source, count=count
+                    )
+            else:
+                yield from process_files_in_folder(subfolder, source)
+
+
+def main(args):
+    base_folder = Path(args.base_folder)
+    to_dolma(
+        examples=process_folder(base_folder, count=args.count),
+        path=args.output_dir,
+        shard_size=args.shard_size,
+        filename="ukhansard.jsonl.gz",
```

Comment:
This should probably be an argument like the CA version, and probably be `uk_hansard.jsonl.gz` to be a bit clearer.